packages/pcd/pod-pcd/src/PODPCD.ts

```diff
@@ -0,0 +1,247 @@
+ * and deriving an {@link PODPCDClaim} from the given {@link PODPCDArgs}.
+ *
+ * @throws if the arguments are invalid
+ */
+export async function prove(args: PODPCDArgs): Promise<PODPCD> {
+  if (!args.privateKey.value) throw new Error("No private key value provided");
+  if (!args.entries.value) throw new Error("No POD entries value provided");
+  const id = typeof args.id.value === "string" ? args.id.value : uuid();
+
+  const pod = POD.sign(args.entries.value, args.privateKey.value);
+  return new PODPCD(id, pod);
+}
+
+/**
+ * Verifies a POD PCD by checking that its {@link PODPCDClaim} corresponds to
+ * its {@link PODPCDProof}.  If the signature is valid and matches the entries,
+ * the function returns true, otherwise false.
+ */
+export async function verify(pcd: PODPCD): Promise<boolean> {
+  try {
+    return pcd.pod.verifySignature();
+  } catch (e) {
+    console.error("Verifying invalid POD data:", e);
+    return false;
+  }
+}
+
+/**
+ * Serializes a {@link PODPCD}.
+ * @param pcd The POD PCD to be serialized.
+ * @returns The serialized version of the POD PCD.
+ */
+export async function serialize(pcd: PODPCD): Promise<SerializedPCD<PODPCD>> {
+  return {
+    type: PODPCDTypeName,
+    pcd: JSONBig({
+      useNativeBigInt: true,
+      alwaysParseAsBig: true
+    }).stringify({
+      id: pcd.id,
+      claim: pcd.claim,
+      proof: pcd.proof
+    })
+  };
+}
+
+/**
+ * Deserializes a serialized {@link PODPCD}.
+ * @param serialized The serialized PCD to deserialize.
+ * @returns The deserialized version of the POD PCD.
+ */
+export async function deserialize(serialized: string): Promise<PODPCD> {
+  const deserialized = JSONBig({
+    useNativeBigInt: true,
+    alwaysParseAsBig: true
+  }).parse(serialized);
+
+  // TODO(artwyman): More careful schema validation, likely with Zod, with
+  // special handling of the PODEntries type and subtypes.
+  // TODO(artwyman): Backward-compatible schema versioning.
+  requireDefinedParameter(deserialized.id, "id");
+  requireDefinedParameter(deserialized.claim, "claim");
+  requireDefinedParameter(deserialized.claim.entries, "entries");
+  requireDefinedParameter(
+    deserialized.claim.signerPublicKey,
+    "signerPublicKey"
+  );
+  requireDefinedParameter(deserialized.proof, "proof");
+  requireDefinedParameter(deserialized.proof.signature, "signature");
+
+  const loadedPOD = POD.load(
+    deserialized.claim.entries,
+    deserialized.proof.signature,
+    deserialized.claim.signerPublicKey
+  );
+
+  return new PODPCD(deserialized.id, loadedPOD);
+}
+
+/**
+ * Provides the information about the {@link PODPCD} that will be displayed
+ * to users on Zupass.
+ * @param pcd The POD PCD instance.
+ * @returns The information to be displayed, specifically `header` and `displayName`.
+ */
+export function getDisplayOptions(
+  // TODO(artwyman): Figure out why this is the only case where using PODPCD directly doesn't work
```

Comment:
Nope, I was waiting for your suggestions, and then got pulled off to focus on ARG rather than PCD SDK use cases.  Let me know if you can help suss this out.

Reply:
@ichub I just put your name on this TODO rather than mine as I think you own the next step here.  It's really a general question: Do we want to support PCD types having fields (data, functions) in addition to `claim` and `proof`, and if so what's the right pattern for doing so?
This isn't going to block the further development of POD PCDs, and thus I don't think it should block this PR, but it would be good to resolve.